File: src/main/java/org/dukcode/ps/codetree/trail02/chapter03/lesson02/IntroNextLevel.java
package org.dukcode.ps.codetree.trail02.chapter03.lesson02;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-next-level/description">LINK</a>
 */
public class IntroNextLevel {

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    Character c1 = new Character("codetree", 10);
    st = new StringTokenizer(br.readLine());
    Character c2 = new Character(st.nextToken(), Integer.parseInt(st.nextToken()));

    bw.write(c1.toString());
    bw.newLine();
    bw.write(c2.toString());

    br.close();
    bw.close();

  }

  private static class Character {

    private String id;
    private int level;

    public Character(String id, int level) {
      this.id = id;
      this.level = level;
    }

    @Override
    public String toString() {
      return "user " + id + " lv " + level;
    }
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter03/lesson02/IntroNextLevel.kt
package org.dukcode.ps.codetree.trail02.chapter03.lesson02.introNextLevel

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-next-level/description">LINK</a>
 */
fun main() {
    val user1 = User("codetree", 10)

    val st = StringTokenizer(br.readLine())
    val id = st.nextToken()
    val level = st.nextToken().toInt()

    val user2 = User(id, level)

    bw.write(user1.toString())
    bw.newLine()
    bw.write(user2.toString())
    bw.newLine()

    bw.flush()
}

class User(
    private val id: String,
    private val level: Int,
) {
    override fun toString() = "user $id lv $level"
}
